### ISSUE TITLE
Mutation Warning: Can't call setState (or forceUpdate) on an unmounted component.

### ISSUE THREAD

**Issue by bicstone** [marmelab/react-admin#3992]
**What happened instead:**
Include `{refresh: true}` and `{undoable: false}` `options` prop to `<Mutation>`.
Click the button, the following error message:

>Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.

**Related code:**

* **[CodeSandbox (click "CHANGE")](https://codesandbox.io/s/simple-jhco0?expanddevtools=1&hidenavigation=1&module=%2Fsrc%2Fposts%2FPostList.js)**
* [Document](https://marmelab.com/react-admin/Actions.html#query-and-mutation-components)

```jsx
<Mutation
  type="UPDATE"
  resource="comments"
  payload={{ id: record.id, data: { ...record, is_approved: true } }}
  options={{
    undoable: false,
    onSuccess: {
      notification: { body: 'Comment approved', level: 'info' },
      refresh: true
    },
    onFailure: {
      notification: { body: 'Error: comment not approved', level: 'warning' },
      refresh: true
    }
  }}
>
  {approve => <Button onClick={approve}>Approve</Button>}
</Mutation>
```

**Environment**

* React-admin version: 2.9.8
* Last version that did not exhibit the issue (if applicable):
* React version: 16.12.0
* Browser: Google Chrome 78.0.3904.97（Official Build）
* Stack trace (in case of a JS error):
```
Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in Mutation (created by Connect(Mutation))
    in Connect(Mutation) (at App.js:26)
    in ApproveButton (at App.js:42)
    in td (created by TableCell)
    in TableCell (created by WithStyles(TableCell))
    in WithStyles(TableCell) (created by DatagridCell)
    in DatagridCell (created by DatagridRow)
    in tr (created by TableRow)
    in TableRow (created by WithStyles(TableRow))
    in WithStyles(TableRow) (created by DatagridRow)
    in DatagridRow (created by Connect(DatagridRow))
    in Connect(DatagridRow)
    in tbody (created by TableBody)
    in TableBody (created by WithStyles(TableBody))
    in WithStyles(TableBody) (created by DatagridBody)
    in DatagridBody (created by shouldUpdate(DatagridBody))
    in shouldUpdate(DatagridBody)
    in table (created by Table)
    in Table (created by WithStyles(Table))
    in WithStyles(Table) (created by Datagrid)
    in Datagrid (created by WithStyles(Datagrid))
    in WithStyles(Datagrid) (at App.js:39)
    in div
```

**Comment by fzaninotto** [comment]
Thanks, reproduced and confirmed. It is fixed in 3.0, though.

**Comment by fzaninotto** [comment]
Fixed by #4012
